cmd/rotate.go

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is subject to the terms and conditions defined in
+ * file 'LICENSE.md', which is part of this source code package.
+ */
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	"github.com/unidoc/unicli/pdf"
+)
+
+const rotateCmdDesc = `Rotate PDF file pages by a specified angle.
+The angle argument is specified in degrees and it must be a multiple of 90.
+
+The command can be configured to rotate only the specified pages
+using the --pages parameter.
+
+An example of the pages parameter: 1-3,4,6-7
+Only pages 1,2,3 (1-3), 4 and 6,7 (6-7) will be rotated, while
+page number 5 is skipped.
+`
+
+var rotateCmdExample = fmt.Sprintf("%s\n%s\n%s\n%s\n%s\n",
+	fmt.Sprintf("%s rotate input_file.pdf 90", appName),
+	fmt.Sprintf("%s rotate -- input_file.pdf -270", appName),
+	fmt.Sprintf("%s rotate -o output_file.pdf input_file.pdf 90", appName),
+	fmt.Sprintf("%s rotate -o output_file.pdf -P 1-3 input_file.pdf 90", appName),
+	fmt.Sprintf("%s rotate -o output_file.pdf -P 1-3 -p pass input_file.pdf 90", appName),
+)
+
+// rotateCmd represents the rotate command
+var rotateCmd = &cobra.Command{
+	Use:                   "rotate [FLAG]... INPUT_FILE ANGLE",
+	Short:                 "Rotate PDF file pages",
+	Long:                  rotateCmdDesc,
+	Example:               rotateCmdExample,
+	DisableFlagsInUseLine: true,
+	Run: func(cmd *cobra.Command, args []string) {
+		// Parse input parameters.
+		inputPath := args[0]
+		password, _ := cmd.Flags().GetString("password")
+
+		// Parse angle parameter.
+		angle, err := strconv.Atoi(args[1])
+		if err != nil {
+			printUsageErr(cmd, "Invalid rotation angle specified\n")
+		}
+
+		// Parse output file.
+		outputPath, _ := cmd.Flags().GetString("output-file")
+		if outputPath == "" {
+			outputPath = inputPath
+		}
+
+		// Parse page range.
+		pageRange, _ := cmd.Flags().GetString("pages")
+
```

Comment:
remove newline?  The statements seem grouped together.  (subjective)

Reply:
For some reason, it does not look at all great to me when there are two regular statements and then an if statement, with no empty lines in between. That's why, I always prefer to force a comment, add the first statement and a new line and then group the second with the if statement. That's because the second statement is related to the if statement. The first however, is not. I found that this scenario does happen quite often in Go.
Anyway, it's totally subjective so I'm not opposed to removing the empty line. Let me know.